### PR TITLE
Use feature flags for signup flow

### PR DESCRIPTION
**Pull request by rottencandy**
## Fixes 
https://issues.redhat.com/browse/HAC-1979

## Description
Use feature flags for detecting user signup status.

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 
(No UI changes)

## Browser conformance: 

- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/components/NamespacedPage/NamespacedPage.tsx [hunk 1]
```diff
@@ -24,24 +25,22 @@ type NamespacedPageProps = {
...
-  if (!namespaceLoaded) {
+  if (!namespaceLoaded || signupFlag === undefined) {
     return (
       <Bullseye>
         <Spinner />
       </Bullseye>
     );
   }
 
+  if (!signupFlag || signupPendingFlag) {
```
Comment (rohitkrai03) [inline src/components/NamespacedPage/NamespacedPage.tsx:40]: Did you try using the feature flags in the route declaration?

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/6041994/185139523-4c70f61a-562d-42d1-aa22-ce43f7d1234d.mov

Singup page is not loading for me if I try with a non singed up user. Also the navigation items should be updated accordingly. It should only render `Singup` nav item if user is not signed up.

**Review by rohitkrai03 — APPROVED** [review]
/approve
/lgtm

**Comment by rottencandy** [comment]
E2e passing locally
screenshot-1661250724

**Pull request merged**